grammars/javascript.cson

```diff
@@ -281,240 +281,283 @@
+            'captures':
+              '1':
+                'name': 'meta.delimiter.method.period.js'
+              '2':
+                'name': 'entity.name.function.js'
+              '3':
+                'name': 'keyword.operator.assignment.js'
+          }
+          {
+            'include': '#function_innards'
+          }
+        ]
+      }
+      {
+        'include': '#function_body'
+      }
+      {
+        'include': '#comments'
+      }
+    ]
   }
   {
-    'begin': '([a-zA-Z_?.$][\\w?.$]*)\\.([a-zA-Z_?.$][\\w?.$]*)\\s*(=)\\s*(?:(async)(?:\\s+))?(function\\*?)\\s*(\\*?)\\s*([a-zA-Z_?$][\\w?$]*)?\\s*(\\()'
-    'beginCaptures':
-      '1':
-        'name': 'support.class.js'
-      '2':
-        'name': 'entity.name.function.js'
-      '3':
-        'name': 'keyword.operator.assignment.js'
-      '4':
-        'name': 'storage.modifier.js'
-      '5':
-        'name': 'storage.type.function.js'
-      '6':
-        'name': 'storage.type.function.js'
-      '7':
-        'name': 'entity.name.function.js'
-      '8':
-        'name': 'punctuation.definition.parameters.begin.js'
-    'comment': 'match stuff like: Sound.play = function() { … }'
-    'end': '(\\))'
-    'endCaptures':
-      '1':
-        'name': 'punctuation.definition.parameters.end.js'
-    'name': 'meta.function.js'
+    # foo: function...
+    'begin': '(?=\\b[a-zA-Z_$][\\w$]*\\s*:\\s*(\\basync\\b\\s*)?\\bfunction\\b)'
+    'end': '(?<=})'
     'patterns': [
       {
-        'include': '#function-params'
+        'begin': '\\G'
+        'end': '(?<=\\))'
+        'name': 'meta.function.json.js'
+        'patterns': [
+          {
+            'match': '\\b([a-zA-Z_$][\\w$]*)\\s*(:)\\s*'
+            'captures':
+              '1':
+                'name': 'entity.name.function.js'
+              '2':
+                'name': 'keyword.operator.assignment.js'
+          }
+          {
+            'include': '#function_innards'
+          }
+        ]
+      }
+      {
+        'include': '#function_body'
+      }
+      {
+        'include': '#comments'
       }
     ]
   }
   {
-    'begin': '([a-zA-Z_?$][\\w?$]*)\\s*(=)\\s*(?:(async)(?:\\s+))?(function\\*?)\\s*(\\*?)\\s*([a-zA-Z_?$][\\w?$]*)?\\s*(\\()'
-    'beginCaptures':
-      '1':
-        'name': 'entity.name.function.js'
-      '2':
-        'name': 'keyword.operator.assignment.js'
-      '3':
-        'name': 'storage.modifier.js'
-      '4':
-        'name': 'storage.type.function.js'
-      '5':
-        'name': 'storage.type.function.js'
-      '6':
-        'name': 'entity.name.function.js'
-      '7':
-        'name': 'punctuation.definition.parameters.begin.js'
-    'comment': 'match stuff like: play = function() { … }'
-    'end': '(\\))'
-    'endCaptures':
-      '1':
-        'name': 'punctuation.definition.parameters.end.js'
-    'name': 'meta.function.js'
+    # "foo": function...
+    'begin': '(?=((\'[^\']*?\')|("[^"]*?"))\\s*:\\s*(\\basync\\b\\s*)?\\bfunction\\b)'
```

Comment:
So from some testing with other grammars I think the following will also work: `((\'.*?\')|(".*?"))`